src/middleware/next.rs

```diff
@@ -1,17 +1,17 @@
-use std::collections::VecDeque;
+use smallvec::SmallVec;
 use std::sync::Arc;
 
 use super::middleware::{BoxFuture, Middleware};
 use crate::error::Error;
 use crate::request::Request;
 
 pub struct Next<T = ()> {
-    deque: VecDeque<Arc<dyn Middleware<T>>>,
```

Comment:
For the sake of correctness, I think I'm going to drop this commit and preallocate for a VecDeque even though it's less performant.

In theory, there isn't anything wrong with this optimization. However, it does feel like implementing a middleware stack cache similar to Actix is a better approach. I prefer to an extra allocation until we optimize the right way.